File: python/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = dict()
        for i, num in enumerate(nums):
            res = target - num
            if res in hashMap:
                return [i, hashMap.get(res)]
            else:
                hashMap[num] = i
        return None
# @lc code=end

File: cpp/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */
#include <unordered_map>
#include <vector>

using namespace std;
// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;
        for (int i = 0; i < nums.size(); i++) {
            int other_one = target - nums[i];
            auto found = hashmap.find(other_one);
            if (found != hashmap.end()) {
                return {i, found->second};
            }
            hashmap.insert(make_pair(nums[i], i));
        }
        return {};
    }
};
// @lc code=end

